### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by gagc056**

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+
+    if user&.authenticate(params[:session][:password])
+      # Log the user in and redirect to the user's show page.
+      cookies.permanent[:user_id] = user.id
+      cookies.permanent[:remember_token] = user.remember_token
```
Comment (Ispirett) [inline app/controllers/sessions_controller.rb]: You should be giving the User a new remember token here [requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_method_requirement.png)
eg:
```
 remember_token = User.generate_remember_token
  user.update_attribute(:remember_token, remember_token)
  cookies.permanent[:remember_token] = remember_token

```

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      flash[:success] = 'Welcome!'
+    else
+      # Create an error message.
+      flash[:error] = 'Wrong combination'
+    end
+    redirect_to signup_path
+  end
+
+  def destroy
```
Comment (Ispirett) [inline app/controllers/sessions_controller.rb:24]: This method should set the` @current_user to nil` [requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/session_delete_requirement.png)

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+Rails.application.routes.draw do
+
+  root 'users#new'
+  get 'sessions/new'
+  get 'users/index'
+  get 'posts/new'
```
Comment (Ispirett) [inline config/routes.rb]: You can remove this

File: config/routes.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+Rails.application.routes.draw do
+
+  root 'users#new'
+  get 'sessions/new'
+  get 'users/index'
```
Comment (Ispirett) [inline config/routes.rb]: this  can also be removed

**Review by Ispirett — CHANGES_REQUESTED** [review]
## Status Changes required :1st_place_medal: :+1: 
Great Work so far @lhscar56  @SantiagoGuerra :+1:  :+1: 
There just some things to take care of below.


There are some issues to address.

- [x] When trying to save this user there is and error but it' not shown to the user. The data is be rollback back and not saving to the database. The issue seems to be the name validation please show a flash message to the user. you can use  @modelname.errors.full_messages if you wish.
```

Name 
tom

Email 
n@gmail.com

Password 
12345678

Password confirmation 
12345678

```
- [x] The user that posted the post is not showing when a user is logged in. [requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png)

- [x] [Optional] when on the /signup path it still shows the login form.

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      flash[:error] = 'Wrong combination'
+      redirect_to signup_path
+    end
+  end
+
+  def destroy
+    cookies.delete :user_id
+    cookies.delete :remember_token
+
+    redirect_to signup_path
```
Comment (samgaco) [inline app/controllers/sessions_controller.rb:28]: As said by @Ispirett 

> This method should set the @current_user to nil [requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/session_delete_requirement.png)

Please, read the requirement in the odin project in [this screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/session_delete_requirement.png)

File: app/controllers/sessions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+
+    if user&.authenticate(params[:session][:password])
+      # Log the user in and redirect to the user's show page.
+      cookies.permanent[:user_id] = user.id
+      cookies.permanent[:remember_token] = user.remember_token
```
Comment (samgaco) [inline app/controllers/sessions_controller.rb]: as said by @Ispirett 
> You should be giving the User a new remember token here [requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_method_requirement.png)
> eg:
> 
> ```
>  remember_token = User.generate_remember_token
>   user.update_attribute(:remember_token, remember_token)
>   cookies.permanent[:remember_token] = remember_token
> ```


You can check the requirement in this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_method_requirement.png)

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @SantiagoGuerra and @lhscar56 :high_brightness: 

Everything looks fine but there are still two points from the previous review left to cover. I left some comments below addressing them.

Regards!

----------------------------

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

File: app/controllers/sessions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    else
+      # Create an error message.
+      flash[:error] = 'Wrong combination'
+      redirect_to signup_path
+    end
+  end
+
+  def destroy
+    cookies.delete :user_id
+    cookies.delete :remember_token
```
Comment (mcrd25) [inline app/controllers/sessions_controller.rb:26]: [OPTIONAL]
You may even consider setting the user's remember_token in DB to nil for security purposes.

**Review by mcrd25 — APPROVED** [review]
# Project Approved. 🎊 ㊗️ 🎉 🤓 

Hi Team!

We are finally at the finish line 🏁! 👍 

I see no need for further review!

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Pull request merged**